server.go

```diff
@@ -1286,6 +1243,12 @@ func (s *Server) applyDeleteShardGroup(m *messaging.Message) (err error) {
 		s.stats.Add("shardsDeleted", int64(len(g.Shards)))
 		return tx.saveDatabase(db)
 	})
+
+	// remove from lookups.
+	for _, sh := range g.Shards {
```

Comment:
This "lookup" map has caused me problems before. We should think about seeing if there is a way to solve the problems it is solving in another way...later, of course.